scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditDatasetController.java

```diff
@@ -0,0 +1,72 @@
+package edu.asu.diging.scriptoocloud.web;
+
+import edu.asu.diging.scriptoocloud.core.model.impl.Dataset;
+import edu.asu.diging.scriptoocloud.core.service.IDatasetService;
+import edu.asu.diging.scriptoocloud.web.forms.DatasetEditForm;
+import edu.asu.diging.scriptoocloud.web.forms.DatasetForm;
+import edu.asu.diging.simpleusers.core.model.IUser;
+import edu.asu.diging.simpleusers.core.service.IUserManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+@Controller
+public class EditDatasetController {
+
+    private final IDatasetService iDatasetService;
+    private final IUserManager userManager;
+
+    @Autowired
+    public EditDatasetController(IDatasetService iDatasetService, IUserManager userManager) {
+        this.iDatasetService = iDatasetService;
+        this.userManager = userManager;
+    }
+
+    @RequestMapping(value = "datasets/{id}/edit", method = RequestMethod.POST)
+    public String edit(@PathVariable("id") Long datasetId,
+                       @Valid @ModelAttribute("datasetEditForm") DatasetEditForm datasetEditForm,
+                       BindingResult result,
+                       RedirectAttributes redirectAttributes,
+                       Model model,
+                       @RequestParam("page") Optional<Integer> page,
+                       @RequestParam("size") Optional<Integer> size,
+                       Principal principal) {
+        // Duplicate Dataset names are okay, but editing a Dataset to have the same name
+        // that it already has (same id, same name) is an error.
```

Comment:
that seems counterintuitive to me. I would just let users save the same information if they want to.

Reply:
In can actually be handy in cases where saving triggers something and you want to test this trigger. You don't have to change something, you just click save.